[institutional] pipeline cli to adapt with institutional config [GEN-1260][GEN-1335]

ochaloup opened the pull request:
A follow-up to #186 

* Adjusting settlement on-chain creation to work with institutional settlements (and Bond config of institutions)
* Adjust verify settlement handling to cover the institutional config pubkey

ochaloup commented:
@martinkrecek and I'm here with again a bit longer PR. It contains refactoring (naming change for cli package that loads on-chain bonds data) and then adapting the API/store functionality to work with other than the default config bonds account.

After merging this PR (and in case some fixes on buildkite processing but I have been testing that so hopefully not many) I believe the base institutional integration could be considered done.

martinkrecek commented on .buildkite/fund-settlements.yml:
```diff
@@ -192,22 +182,21 @@ steps:
         "attachments": [{
           "color": "'"$$notification_color"'",
-          "title": "Fund Settlements '"$${notification_result}$${claim_type}"', '"$$attempts_count"' attempts",
+          "title": "Fund '"$$claim_type Settlements"' '"$${notification_result}"', '"$$attempts_count"' attempts",
```
(minor non-blocking note - consistently with other pipeline files)
```suggestion
          "title": "Fund '"$$claim_type"' Settlements '"$${notification_result}"', '"$$attempts_count"' attempts",
```

martinkrecek reviewed (commented):
There's a conflict with the base branch, please resolve it before my re-review.

ochaloup commented on README.md:
```diff
@@ -12,13 +12,14 @@ Mono repository for Validator Bonds product
 * [`migrations/`](./migrations/) - SQL scripts to prepare and change DB schemas
 * [`scripts/`](./scripts/) - scripts used in pipeline and to manage and integrate various repository parts
-* [`validator-bonds-cli`](./validator-bonds-cli/) - CLI used by operator to manage bonds (user related CLI is [packages/CLI](./packages/validator-bonds-cli/))
+* [`validator-bonds-cli`](bonds-collector/) - CLI used by operator to manage bonds (user related CLI is [packages/CLI](./packages/validator-bonds-cli/))
```
That's a duplication. Thanks. I will remove it.

ochaloup commented on api/src/bin/api.rs:
```diff
@@ -94,7 +94,20 @@ async fn main() -> anyhow::Result<()> {
         .and(with_context(context.clone()))
-        .and_then(bonds::handler);
+        .and_then(bonds::handler)
+        .or(warp::path!("bonds" / "bidding")
```
The point is to assign the same handler for "bonds/bidding" and "bonds". As far as I was testing this, it was working that way.

I'm fine to split it into two separate routes. If you consider that better practice, easier to read or whatever. Can you share with me your preference on this please?

martinkrecek replied:
If the point is to assign the same handler for both paths, then that doesn't match what's happening in the code, because there is `.and_then(bonds::handler)` and also `.and_then(bonds::handler_bidding))`.

Reusing the handler is exactly what I was expecting when seeing that it's tied together, but literally all the `.and(...)` calls are duplicated for both, so I realized that I don't understand what it's supposed to achieve and that it might be somehow missing the point.

To be clear, if the paths are built correctly then this doesn't matter functionality-wise, I just wanted to understand why this approach was chosen, because it seems like it's going somewhere (reusing the handler) but failing along the way (copying everything and using a different handler).

ochaloup commented on .buildkite/claim-settlements.yml:
```diff
@@ -8,31 +8,57 @@ env:
+        key: "claim_type"
+        options:
+          - label: "Bidding Settlements"
+            value: "bid"
```
That's right, but there is an issue in the naming of the file as the value is used for that purpose as well. So a bit of inconsistency, sorry, I'm probably not able to fix that "anyhow correctly" now. I will add at least a comment here to explain.

ochaloup replied:
I was thinking of fixing it in the opposite direction (to use the term `bid` at all places), but unfortunately `bidding` is used in DB already. Now we have bidding in DB and `bid` in the file system.

Maybe the `claim_type` should be really `bidding` at all places, and only to be replaced with `bid` in the pipeline later.

ochaloup replied:
Ah, this happens to be complicated to do consistently. At other places we need to differentiate not only on type of bid vs. institutional but on type of bid. And as mentioned that type is already used at filesystem.
I'm going to make consistent `collect-bonds.yml` but otherwise leaving it as it is.

ochaloup commented on .buildkite/claim-settlements.yml:
```diff
@@ -8,31 +8,57 @@ env:
   - label: ":closed_lock_with_key: Concurrency gate lock"
     command: echo "--> Start of concurrency gate"
-    concurrency_group: 'validator-bonds/claim-settlements'
+    concurrency_group: 'validator-bonds/claim-settlements-${CLAIM_TYPE}'
```
Yes, I'm aware of it and as the metadata is not possible to be part of the concurrency group, it may by run in two instances. The pipeline does not need to be run multiple times (no data consistency issues are expected in that case). The concurrency group is here more to avoid running many pipelines when the first one gets stuck with an issue.

Plus. My thinking was following - the `CLAIM_TYPE` env var is defined in cron schedule (triggering the job automatically). When the CLAIM_TYPE is not set then it is probably a run manually trigerred manually from UI. The manual run is kind of "expected" to be running despite a scheduled one that has got stuck.

ochaloup commented:
@martinkrecek thanks for the review. I tried to address your points, fixed the merge conflict and added two small updates on buildkite pipelines to correct them.

The PR is ready for re-review.

ochaloup commented:
Ah. I missed your point about the handler. I need to fix that as an additional point.

ochaloup commented:
@ochaloup Sorry for skipping around, now it is ready for review.

ochaloup commented:
@martinkrecek  Because of PR#198, the underlying pipelines have changed, and I need to fix merge conflicts. Would you be so kind as to re-review this PR?
